docs/source/tutorials/fine_tune_llama_7b.mdx

```diff
@@ -0,0 +1,356 @@
+
+We can click on it, and a **`jupyter`** environment opens in our local browser. Open the notebook **`llama2-7b-fine-tuning.ipynb`** and lets get started.
+
+_Note: We are going to use the Jupyter environment only for preparing the dataset and then `torchrun` for launching our training script for distributed training._
+
+If you are going to use official Llama 2 checkpoint you need to login into our hugging face account, which has access to the model, to use your token for accessing the gated repository. We can do this by running the following command:
+
+_Note: We also provide an ungated checkpoint._
+
+```python
+!huggingface-cli login --token YOUR_TOKEN
+```
+
+## 2. Load and prepare the dataset
+
+We will use [Dolly](https://huggingface.co/datasets/databricks/databricks-dolly-15k) an open source dataset of instruction-following records on categories outlined in the [InstructGPT paper](https://arxiv.org/abs/2203.02155), including brainstorming, classification, closed QA, generation, information extraction, open QA, and summarization.
+
+```python
+{
+  "instruction": "What is world of warcraft",
+  "context": "",
+  "response": "World of warcraft is a massive online multi player role playing game. It was released in 2004 by bizarre entertainment"
+}
+```
+
+To load the `dolly` dataset, we use the `load_dataset()` method from the 🤗 Datasets library.
+
+```python
+from datasets import load_dataset
+from random import randrange
+
+# Load dataset from the hub
+dataset = load_dataset("databricks/databricks-dolly-15k", split="train")
+
+print(f"dataset size: {len(dataset)}")
+print(dataset[randrange(len(dataset))])
+# dataset size: 15011
+
+```
+
+To instruct tune our model we need to convert our structured examples into a collection of tasks described via instructions. We define a `formatting_function` that takes a sample and returns a string with our format instruction.
+
+```python
+def format_dolly(sample):
+    instruction = f"### Instruction\n{sample['instruction']}"
+    context = f"### Context\n{sample['context']}" if len(sample["context"]) > 0 else None
+    response = f"### Answer\n{sample['response']}"
+    # join all the parts together
+    prompt = "\n\n".join([i for i in [instruction, context, response] if i is not None])
+    return prompt
+
+```
+
+lets test our formatting function on a random example.
+
+```python
+from random import randrange
+
+print(format_dolly(dataset[randrange(len(dataset))]))
+```
+
+In addition, to formatting our samples we also want to pack multiple samples to one sequence to have a more efficient training. This means that we are stacking multiple samples to one sequence and split them with an EOS Token. This makes the training more efficient. Packing/stacking samples can be done during training or before. We will do it before training to save time. We created a utility method [pack_dataset](./scripts/utils/pack_dataset.py) that takes a dataset and a packing function and returns a packed dataset.
```

Comment:
You might want to update the link, as it is relative to the notebooks directory.

```suggestion
In addition, to formatting our samples we also want to pack multiple samples to one sequence to have a more efficient training. This means that we are stacking multiple samples to one sequence and split them with an EOS Token. This makes the training more efficient. Packing/stacking samples can be done during training or before. We will do it before training to save time. We created a utility method [pack_dataset](https://github.com/huggingface/optimum-neuron/tree/main/notebooks/text-generation/scripts/utils/pack_dataset.py) that takes a dataset and a packing function and returns a packed dataset.
```